feat: integrate NCA exclusion flags (NCAwXRS) in the mapping

Gero1999 opened the pull request:
## Issue

Contributes to #752 

## Description

Flags for NCA exclusion (NCAwXRS) are incorporated in the mapping as a selectInput. It allows multiple mapping and uses PKNCA system to do exclusions (that means the records won't be filtered, so the data is not changed)

The dummy data also has now two flag columns that are flagged with:

- `Invalid administration` for patients 2 & 4 in DOSES 1 & 2
- `No AVAL value` for a missing value that was manually added. Records at rows 100 & 233
- In SUBJ2 you can also find a case in which both exclusion reasons are happening

## Definition of Done
- [x] NCAwXRS exclusion flags in input and CDISC outputs (ADNCA)

## How to test

- Map data (default) > Run NCA > Check that patients 2 has no DOSE 2 results

-  Map data (without NCA1XRS) > Run NCA > Check that patients 2 has DOSE 2 results


- Check the ADNCA CDISC files contains the columns that were used in the mapping (if you don't include it in the mapping, they should not be)! And of course they always should use CDISC nomenclature, not the user's one


## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented
- [x] App or package changes are reflected in NEWS
- [x] Package version is incremented

Gero1999 committed:
feat: include NCAwXRS mapping option

Gero1999 committed:
data: add two NCAXFL columns & NA in sample 222

adnca <- read.csv("inst/shiny/data/Dummy_data.csv")
adnca$AVAL[222] <- NA_real_
adnca2 <- adnca %>%
  mutate(NCA1XRS = ifelse(USUBJID %in% unique(USUBJID)[c(2,4)] & ATPTREF == "DOSE 2", "Vomiting", "")) %>%
  mutate(NCA2XRS = ifelse(is.na(AVAL), "Missing AVAL value", ""))
write.csv(adnca2, "inst/shiny/data/Dummy_data.csv", row.names = FALSE)

Gero1999 committed:
feat: integrate with PKNCA & fix create_start_impute issue

Gero1999 committed:
integrate NCA exclusion flags in CDISC ADNCA dataset

Gero1999 committed:
add some docs for the helper fun

Gero1999 committed:
fix: prevent creating NCA exclusion columns when no exclusion is done

Gero1999 committed:
fix labels and split action

Gero1999 committed:
create CDISC tests for the derivation of NCA exclusion flags

Gero1999 committed:
refactor: lintr

Gero1999 committed:
data: add a case with both exclusion reasons

adnca <- read.csv("inst/shiny/data/Dummy_data.csv")
ix <- which(adnca$NCA1XRS != "")[1]
adnca$NCA2XRS[ix] = "Missing AVAL value"
write.csv(adnca, "inst/shiny/data/Dummy_data.csv")

Gero1999 committed:
Merge remote-tracking branch 'origin/main' into 752-feat/exclusion-flags-mapping

Gero1999 committed:
Merge remote-tracking branch 'origin/main' into 752-feat/exclusion-flags-mapping

Shaakon35 reviewed (changes requested):
The zip file does not contain all the columns of the ADNCA dataset displayed in the app, could you please fix this? and have a look at ADPP as well please.

Otherwise, all good!!

js3110 reviewed (commented):
Looks like maybe some merges or smth are adding additional columns:
<img width="647" height="461" alt="image" src="https://github.com/user-attachments/assets/f898ed79-a324-4e5a-897a-e37b2f2c5e48" />

Otherwise looks good, I'll wait for this to be fixed then approve

Gero1999 committed:
merge: rm data file, so no conflict to address

Merge remote-tracking branch 'origin/main' into 752-feat/exclusion-flags-mapping

# Conflicts:
#	inst/shiny/data/Dummy_data.csv

Gero1999 committed:
data: add NCA1XRS & NCA2XRS in example-ADNCA.csv

adnca <- read.csv("inst/shiny/data/example-ADNCA.csv")
adnca$AVAL[c(100, 233)] <- NA_real_
adnca2 <- adnca %>%
  mutate(NCA1XRS = ifelse(USUBJID %in% unique(USUBJID)[c(2,4)] & ATPTREF == "DOSE 2", "Vomiting", "")) %>%
  mutate(NCA2XRS = ifelse(is.na(AVAL), "Missing AVAL value", ""))
write.csv(adnca2, "inst/shiny/data/example-ADNCA.csv", row.names = FALSE)

Gero1999 commented:
@js3110 the problem was in the dataset, now with the merge of the new ADNCA hopefully is fixed

js3110 reviewed (commented):
Thanks for the updates, but now it looks like all subjects have NCA results even if the entire profile has NCA1XRS. Is this correct? In the how to test you said it shouldn't have results but maybe its different with the new data

Gero1999 committed:
data: change reasons

adnca <- read.csv("inst/shiny/data/example-ADNCA.csv")
adnca$AVAL[c(100, 233)] <- NA_real_
adnca2 <- adnca %>%
  mutate(NCA1XRS = ifelse(USUBJID %in% unique(USUBJID)[c(2,4)] & NFRLT <= 48, "Invalid administration", "")) %>%
  mutate(NCA2XRS = ifelse(is.na(AVAL), "Missing AVAL value", ""))
write.csv(adnca2, "inst/shiny/data/example-ADNCA.csv", row.names = FALSE)

js3110 reviewed (approved):
lgtm!

Gero1999 committed:
news: NCA flag rules option 

Added option to include NCA flag rules as per ADNCA standards.

Gero1999 committed:
Update package version to 0.1.0.9011

Gero1999 committed:
Merge branch 'main' into 752-feat/exclusion-flags-mapping

Gero1999 committed:
update wordlist
